crates/utils/src/tracing/mod.rs

```diff
@@ -1,6 +1,73 @@
 pub mod grpc;
 
-// Re-export useful traits for open-telemetry traces. This avoids requiring other crates from
-// importing that family of crates directly.
-pub use opentelemetry::trace::Status as OtelStatus;
-pub use tracing_opentelemetry::OpenTelemetrySpanExt;
+use core::time::Duration;
+
+use miden_objects::{block::BlockNumber, Digest};
+use opentelemetry::{trace::Status as OtelStatus, Key, Value};
+use sealed::sealed;
+
+/// Utility functions for converting types into [`opentelemetry::Value`].
+pub trait ToValue {
+    fn to_value(&self) -> Value;
+}
+
+impl ToValue for Duration {
+    fn to_value(&self) -> Value {
+        self.as_secs_f64().into()
+    }
+}
+
+impl ToValue for Digest {
+    fn to_value(&self) -> Value {
+        self.to_hex().into()
+    }
+}
+
+impl ToValue for f64 {
+    fn to_value(&self) -> Value {
+        (*self).into()
+    }
+}
+
+impl ToValue for BlockNumber {
+    fn to_value(&self) -> Value {
+        i64::from(self.as_u32()).into()
+    }
+}
+
+impl ToValue for u32 {
+    fn to_value(&self) -> Value {
+        i64::from(*self).into()
+    }
+}
+
+impl ToValue for i64 {
+    fn to_value(&self) -> Value {
+        (*self).into()
+    }
+}
```

Comment:
@Mirko-von-Leipzig LMK impls to refactor or delete here. Also LMK where you would move / modularize any of this. I have just lumped it in mod.rs for now.

Reply:
I think maybe move all this into a `span_ext.rs` module/file.

We could also consider doing a blanket implementation, although maybe being explicit is fine too.

```rust
impl<T> ToValue for T where T: Into<Value>
```

Reply:
I did try this at first but it causes implementation conflicts for the types that don't impl Into<Value> so don't think it works here.